### PR TITLE
Work towards updating to Kitchen-Terraform 5

### PR DESCRIPTION
**Pull request by tdsacilowski**
Inspec-gcp doesn't seem to be working in Kitchen-Terraform 5.1.0, even when pinning to Inspec 3. Working with @ncs-alane to resolve. See https://github.com/newcontext-oss/kitchen-terraform/issues/351.

### CODE REVIEW

**Comment by tdsacilowski** [comment]
Gist of debug output: https://gist.github.com/tdsacilowski/1ac791085ad9d1c582725543a0ecefa5

**Comment by tdsacilowski** [comment]
For context, pinning to Inspec 3 because of https://github.com/inspec/inspec-gcp/issues/153 but looks like that might be resolved soon.

**Comment by aaron-lane** [comment]
I tested Kitchen-Terraform 5 with InSpec 4 using https://github.com/terraform-google-modules/terraform-google-module-template with an updated Gemfile and I was able to use inspec-gcp resources which indicates the inspec-gcp on InSpec 3 is not actually at play.

**Comment by aaron-lane** [comment]
The error message is coming from train when it tries to load `train/transports/gcp`, https://github.com/inspec/train/blob/v2.1.14/lib/train.rb#L39-L62, but that file is defined by train, https://github.com/inspec/train/blob/v2.1.14/lib/train/transports/gcp.rb, so there may be an error occurring when that file is loaded on your system.

**Comment by tdsacilowski** [comment]
So strange... I even went so far as to spin up a clean Ubuntu 18.04 Vagrant box and I'm running into the same there, so it doesn't seem to be related to my specific environment.

I also created a simple module using https://github.com/terraform-google-modules/terraform-google-module-template, updated the `Gemfile` as such:

```
ruby '2.5.3'

source 'https://rubygems.org/' do
    gem 'kitchen-terraform', '~> 5.1'
end
```

And had the same outcome: ` Message: simple_example gcp: Can't find train plugin gcp. Please install it first.`

@aaron-lane, am I maybe missing a step in setting up my environment somehow? This is what I did on the clean Ubuntu 18.04 Vagrant box:

```bash
# Update packages
sudo apt update && sudo apt -y upgrade

# Install Terraform
sudo apt -y install unzip && \
  curl -O https://releases.hashicorp.com/terraform/0.12.3/terraform_0.12.3_linux_amd64.zip && \
  sudo unzip terraform_0.12.3_linux_amd64.zip -d /usr/local/bin/

# Install rbenv & ruby-build plugin
git clone https://github.com/rbenv/rbenv.git ~/.rbenv
echo -e '\nexport PATH="$HOME/.rbenv/bin:$PATH"' >> ~/.bashrc
echo -e 'eval "$(rbenv init -)"' >> ~/.bashrc
source ~/.bashrc
mkdir -p "$(rbenv root)"/plugins
git clone https://github.com/rbenv/ruby-build.git "$(rbenv root)"/plugins/ruby-build

# Install Ruby 2.5.3
# (pinned version in https://github.com/terraform-google-modules/terraform-google-module-template)
rbenv install 2.5.3
rbenv global 2.5.3

# Install Bundler
gem install bundler

# Install Google SDK
echo "deb [signed-by=/usr/share/keyrings/cloud.google.gpg] https://packages.cloud.google.com/apt cloud-sdk main" | sudo tee -a /etc/apt/sources.list.d/google-cloud-sdk.list
apt-get install apt-transport-https ca-certificates
curl https://packages.cloud.google.com/apt/doc/apt-key.gpg | sudo apt-key --keyring /usr/share/keyrings/cloud.google.gpg add -
sudo apt-get update && sudo apt-get install google-cloud-sdk

# Set up terraform-google-module-template project
cd ~ && git clone https://github.com/terraform-google-modules/terraform-google-module-template
sudo apt -y install python-pip
echo -e '\nexport PATH=$HOME/.local/bin:$PATH' >> ~/.bashrc
source ~/.bashrc
pip install --user cookiecutter
echo -e '\nexport GOOGLE_APPLICATION_CREDENTIALS=/vagrant/credentials.json' >> ~/.bashrc
source ~/.bashrc
```

At this point, I generated a new module, added a `terraform.tfvars` file with `project_id` and `bucket_name`, updated the `Gemfile` for Kitchen-Terraform 5 and did `bundle install`.

`bundle exec kitchen create` and `bundle exec kitchen converge` actions complete successfully, but `bundle exec kitchen verify` fails.

Also, based on https://github.com/inspec/inspec-gcp/pull/155, it seems that inspec-gcp hasn't been updated for InSpec 4 yet, so not sure how it worked for you (unless I misunderstood your comment).

Pinning to `gem 'inspec', '~> 3.0', '>= 3.0.25'` to force InSpec 3 here also doesn't help.

Aaron, were you able to throw what you tested into a repo? I'm curious what's different here. I'm obviously doing something wrong or missing a step somewhere. 🤔

**Comment by nrvale0** [comment]
Is the script executing in the context of /bin/bash or /bin/sh? I've seen rbenv (and indeed anyenv stuff in general) fail to properly load with the use of 'source' because script was executing as /bin/sh and that expects '.' rather than 'source'. It's highly dependent on the sh and BASH implementations. BASH implementation you say?!? Doesn't Ubuntu use dash or something not quite GNU BASH? And is it actually the same binary but behaves different depending on how it was invoked?

**Comment by aaron-lane** [comment]
I'll try to add some reproduction steps tonight

File: README.md [hunk 1]
```diff
@@ -14,35 +14,68 @@ Currently, the only example available in this project if for GCP. The below step
...
+```bash
+# Update packages
+sudo apt update && sudo apt -y upgrade
+
+# Install Terraform
+sudo apt -y install unzip && \
+  curl -O https://releases.hashicorp.com/terraform/0.12.3/terraform_0.12.3_linux_amd64.zip && \
+  sudo unzip terraform_0.12.3_linux_amd64.zip -d /usr/local/bin/
+
+# Install rbenv & ruby-build plugin
```
Comment (aaron-lane) [inline README.md]: Git is not necessarily installed on a base 18.04 image.

```suggestion
# Install rbenv & ruby-build plugin
sudo apt -y install git
```

File: README.md [hunk 2]
```diff
@@ -14,35 +14,68 @@ Currently, the only example available in this project if for GCP. The below step
...
+git clone https://github.com/rbenv/rbenv.git ~/.rbenv
+echo -e '\nexport PATH="$HOME/.rbenv/bin:$PATH"' >> ~/.bashrc
+echo -e 'eval "$(rbenv init -)"' >> ~/.bashrc
+. ~/.bashrc
+mkdir -p "$(rbenv root)"/plugins
+git clone https://github.com/rbenv/ruby-build.git "$(rbenv root)"/plugins/ruby-build
+
```
Comment (aaron-lane) [inline README.md]: Several additional packages are required to install Ruby.

```suggestion
sudo apt -y install build-essential libssl-dev libreadline-dev zlib1g-dev

```

**Review by aaron-lane — CHANGES_REQUESTED** [review]

**Comment by aaron-lane** [comment]
I went through the instructions you added to the README using a GCP VM running ubuntu-1804-bionic-v20190628 and inspec-gcp was successfully loaded. Besides the Vagrant box, in which other environments are you able to reproduce this issue?

**Comment by tdsacilowski** [comment]
@aaron-lane Apologies for the delay in responding here... was on vacation for the past two weeks. 

Had a chance to get back to this today and I'm still running into issues with inspec-gcp. Are you sure that it was actually running the inspec-gcp plugin? Sticking with the https://github.com/terraform-google-modules/terraform-google-module-template for consistency, I had the following happen to me (same setup as described above, just with the prerequisites that you indicated):

```
vagrant@ubuntu-bionic:~/terraform-google-test-module$ bundle exec kitchen verify
-----> Starting Kitchen (v2.2.5)
-----> Setting up <simple-example-default>...
       Finished setting up <simple-example-default> (0m0.00s).
-----> Verifying <simple-example-default>...
$$$$$$ Running command `terraform workspace select kitchen-terraform-simple-example-default` in directory /home/vagrant/terraform-google-test-module/test/fixtures/simple_example
$$$$$$ Running command `terraform output -json` in directory /home/vagrant/terraform-google-test-module/test/fixtures/simple_example
simple_example local: Verifying
Skipping profile: 'inspec-gcp' on unsupported platform: 'ubuntu/18.04'.

Profile: simple_example
Version: (not specified)
Target:  local://

  ×  gcloud: gcloud (3 failed)
     ×  Command: `gcloud --project=teddy-testing services list --enabled` exit_status should eq 0

     expected: 0
          got: 1

     (compared using ==)

     ×  Command: `gcloud --project=teddy-testing services list --enabled` stderr should eq ""

     expected: ""
          got: "ERROR: (gcloud.services.list) You do not currently have an active account selected.\nPlease run:\n\n...t:\n\n  $ gcloud config set account ACCOUNT\n\nto select an already authenticated account to use.\n"

     (compared using ==)

     Diff:
     @@ -1 +1,12 @@
     +ERROR: (gcloud.services.list) You do not currently have an active account selected.
     +Please run:
     +
     +  $ gcloud auth login
     +
     +to obtain new credentials, or if you have already logged in with a
     +different account:
     +
     +  $ gcloud config set account ACCOUNT
     +
     +to select an already authenticated account to use.

     ×  Command: `gcloud --project=teddy-testing services list --enabled` stdout should match "storage-api.googleapis.com"
     expected "" to match "storage-api.googleapis.com"
  ×  gsutil: gsutil (3 failed)
     ×  Command: `gsutil ls -p teddy-testing` exit_status should eq 0

     expected: 0
          got: 1

     (compared using ==)

     ×  Command: `gsutil ls -p teddy-testing` stderr should eq ""

     expected: ""
          got: "ServiceException: 401 Anonymous caller does not have storage.buckets.list access to project 39746844090.\n"

     (compared using ==)

     Diff:
     @@ -1 +1,2 @@
     +ServiceException: 401 Anonymous caller does not have storage.buckets.list access to project 39746844090.

     ×  Command: `gsutil ls -p teddy-testing` stdout should match "gs://simple-example-raptor"
     expected "" to match "gs://simple-example-raptor"


Profile Summary: 0 successful controls, 2 control failures, 0 controls skipped
Test Summary: 0 successful, 6 failures, 0 skipped
>>>>>> ------Exception-------
>>>>>> Class: Kitchen::ActionFailed
>>>>>> Message: 1 actions failed.
>>>>>>     Verify failed on instance <simple-example-default>.  Please see .kitchen/logs/simple-example-default.log for more details
>>>>>> ----------------------
>>>>>> Please see .kitchen/logs/kitchen.log for more details
>>>>>> Also try running `kitchen diagnose --all` for configuration
```

I *thought* that meant that the GCP plugin was loaded since I didn't see any errors related to that but as soon as I commented out the `local` backend test and left only the `gcp` backend test enabled defined:

```
suites:
  - name: simple_example
    driver:
      root_module_directory: test/fixtures/simple_example/
    verifier:
      color: false
      systems:
        - name: simple_example gcp
          backend: gcp
          controls:
            - gcp
```

...I'm back to my original error:

```
...
E, [2019-07-29T18:04:58.953534 #19327] ERROR -- simple-example-default: ----End Backtrace-----
E, [2019-07-29T18:04:58.953542 #19327] ERROR -- simple-example-default: ---Nested Exception---
E, [2019-07-29T18:04:58.953549 #19327] ERROR -- simple-example-default: Class: Kitchen::Terraform::Error
E, [2019-07-29T18:04:58.953557 #19327] ERROR -- simple-example-default: Message: simple_example gcp: Can't find train plugin gcp. Please install it first.
E, [2019-07-29T18:04:58.953568 #19327] ERROR -- simple-example-default: ----------------------
E, [2019-07-29T18:04:58.953576 #19327] ERROR -- simple-example-default: ------Backtrace-------
E, [2019-07-29T18:04:58.953583 #19327] ERROR -- simple-example-default: /home/vagrant/.rbenv/versions/2.5.3/lib/ruby/gems/2.5.0/gems/kitchen-terraform-5.1.0/lib/kitchen/verifier/terraform.rb:109:in `rescue in call'
...
```

This is now what I'm seeing on my Mac and in a clean Ubuntu 18.04 Vagrant box. I'll try some other environments as well.

**Comment by tdsacilowski** [comment]
@aaron-lane Apologies for the delay in responding here... was on vacation for the past two weeks. 

Had a chance to get back to this today and I'm still running into issues with inspec-gcp. Are you sure that it was actually running the inspec-gcp plugin? Sticking with the https://github.com/terraform-google-modules/terraform-google-module-template for consistency, I had the following happen to me (same setup as described above, just with the prerequisites that you indicated):

```
vagrant@ubuntu-bionic:~/terraform-google-test-module$ bundle exec kitchen verify
-----> Starting Kitchen (v2.2.5)
-----> Setting up <simple-example-default>...
       Finished setting up <simple-example-default> (0m0.00s).
-----> Verifying <simple-example-default>...
$$$$$$ Running command `terraform workspace select kitchen-terraform-simple-example-default` in directory /home/vagrant/terraform-google-test-module/test/fixtures/simple_example
$$$$$$ Running command `terraform output -json` in directory /home/vagrant/terraform-google-test-module/test/fixtures/simple_example
simple_example local: Verifying
Skipping profile: 'inspec-gcp' on unsupported platform: 'ubuntu/18.04'.

Profile: simple_example
Version: (not specified)
Target:  local://

  ×  gcloud: gcloud (3 failed)
     ×  Command: `gcloud --project=teddy-testing services list --enabled` exit_status should eq 0

     expected: 0
          got: 1

     (compared using ==)

     ×  Command: `gcloud --project=teddy-testing services list --enabled` stderr should eq ""

     expected: ""
          got: "ERROR: (gcloud.services.list) You do not currently have an active account selected.\nPlease run:\n\n...t:\n\n  $ gcloud config set account ACCOUNT\n\nto select an already authenticated account to use.\n"

     (compared using ==)

     Diff:
     @@ -1 +1,12 @@
     +ERROR: (gcloud.services.list) You do not currently have an active account selected.
     +Please run:
     +
     +  $ gcloud auth login
     +
     +to obtain new credentials, or if you have already logged in with a
     +different account:
     +
     +  $ gcloud config set account ACCOUNT
     +
     +to select an already authenticated account to use.

     ×  Command: `gcloud --project=teddy-testing services list --enabled` stdout should match "storage-api.googleapis.com"
     expected "" to match "storage-api.googleapis.com"
  ×  gsutil: gsutil (3 failed)
     ×  Command: `gsutil ls -p teddy-testing` exit_status should eq 0

     expected: 0
          got: 1

     (compared using ==)

     ×  Command: `gsutil ls -p teddy-testing` stderr should eq ""

     expected: ""
          got: "ServiceException: 401 Anonymous caller does not have storage.buckets.list access to project 39746844090.\n"

     (compared using ==)

     Diff:
     @@ -1 +1,2 @@
     +ServiceException: 401 Anonymous caller does not have storage.buckets.list access to project 39746844090.

     ×  Command: `gsutil ls -p teddy-testing` stdout should match "gs://simple-example-raptor"
     expected "" to match "gs://simple-example-raptor"


Profile Summary: 0 successful controls, 2 control failures, 0 controls skipped
Test Summary: 0 successful, 6 failures, 0 skipped
>>>>>> ------Exception-------
>>>>>> Class: Kitchen::ActionFailed
>>>>>> Message: 1 actions failed.
>>>>>>     Verify failed on instance <simple-example-default>.  Please see .kitchen/logs/simple-example-default.log for more details
>>>>>> ----------------------
>>>>>> Please see .kitchen/logs/kitchen.log for more details
>>>>>> Also try running `kitchen diagnose --all` for configuration
```

I *thought* that meant that the GCP plugin was loaded since I didn't see any errors related to that but as soon as I commented out the `local` backend test and left only the `gcp` backend test enabled defined:

```
suites:
  - name: simple_example
    driver:
      root_module_directory: test/fixtures/simple_example/
    verifier:
      color: false
      systems:
        - name: simple_example gcp
          backend: gcp
          controls:
            - gcp
```

...I'm back to my original error:

```
...
E, [2019-07-29T18:04:58.953534 #19327] ERROR -- simple-example-default: ----End Backtrace-----
E, [2019-07-29T18:04:58.953542 #19327] ERROR -- simple-example-default: ---Nested Exception---
E, [2019-07-29T18:04:58.953549 #19327] ERROR -- simple-example-default: Class: Kitchen::Terraform::Error
E, [2019-07-29T18:04:58.953557 #19327] ERROR -- simple-example-default: Message: simple_example gcp: Can't find train plugin gcp. Please install it first.
E, [2019-07-29T18:04:58.953568 #19327] ERROR -- simple-example-default: ----------------------
E, [2019-07-29T18:04:58.953576 #19327] ERROR -- simple-example-default: ------Backtrace-------
E, [2019-07-29T18:04:58.953583 #19327] ERROR -- simple-example-default: /home/vagrant/.rbenv/versions/2.5.3/lib/ruby/gems/2.5.0/gems/kitchen-terraform-5.1.0/lib/kitchen/verifier/terraform.rb:109:in `rescue in call'
...
```

This is now what I'm seeing on my Mac and in a clean Ubuntu 18.04 Vagrant box. I'll try some other environments as well.

**Comment by tdsacilowski** [comment]
Spun up an Ubuntu 16.04 box, going through the same steps as above and same error:

Gemfile:
```
ruby '2.5.3'

source 'https://rubygems.org/' do
  # gem 'kitchen-terraform', '~> 4.8'
  gem 'kitchen-terraform', '~> 5.1'
end
```

Verify output:
```
vagrant@ubuntu-xenial:~/terraform-google-test-module$ bundle exec kitchen verify
-----> Starting Kitchen (v2.2.5)
-----> Verifying <simple-example-default>...
$$$$$$ Running command `terraform workspace select kitchen-terraform-simple-example-default` in directory /home/vagrant/terraform-google-test-module/test/fixtures/simple_example
$$$$$$ Running command `terraform output -json` in directory /home/vagrant/terraform-google-test-module/test/fixtures/simple_example
simple_example local: Verifying
Skipping profile: 'inspec-gcp' on unsupported platform: 'ubuntu/16.04'.

Profile: simple_example
Version: (not specified)
Target:  local://

  ✔  gsutil: gsutil
     ✔  Command: `gsutil ls -p teddy-testing` exit_status should eq 0
     ✔  Command: `gsutil ls -p teddy-testing` stderr should eq ""
     ✔  Command: `gsutil ls -p teddy-testing` stdout should match "gs://simple-example-ocelot"
  ✔  gcloud: gcloud
     ✔  Command: `gcloud --project=teddy-testing services list --enabled` exit_status should eq 0
     ✔  Command: `gcloud --project=teddy-testing services list --enabled` stderr should eq ""
     ✔  Command: `gcloud --project=teddy-testing services list --enabled` stdout should match "storage-api.googleapis.com"


Profile Summary: 2 successful controls, 0 control failures, 0 controls skipped
Test Summary: 6 successful, 0 failures, 0 skipped
>>>>>> ------Exception-------
>>>>>> Class: Kitchen::ActionFailed
>>>>>> Message: 1 actions failed.
>>>>>>     Verify failed on instance <simple-example-default>.  Please see .kitchen/logs/simple-example-default.log for more details
>>>>>> ----------------------
>>>>>> Please see .kitchen/logs/kitchen.log for more details
>>>>>> Also try running `kitchen diagnose --all` for configuration
```

Error log:
```
vagrant@ubuntu-xenial:~/terraform-google-test-module$ cat .kitchen/logs/simple-example-default.log
I, [2019-08-05T17:55:41.764525 #11375]  INFO -- simple-example-default: -----> Verifying <simple-example-default>...
W, [2019-08-05T17:55:41.765278 #11375]  WARN -- simple-example-default: Running command `terraform workspace select kitchen-terraform-simple-example-default` in directory /home/vagrant/terraform-google-test-module/test/fixtures/simple_example
I, [2019-08-05T17:55:41.998973 #11375]  INFO -- simple-example-default: simple_example local: Verifying
W, [2019-08-05T17:55:42.351916 #11375]  WARN -- simple-example-default: Skipping profile: 'inspec-gcp' on unsupported platform: 'ubuntu/16.04'.
E, [2019-08-05T17:55:44.166002 #11375] ERROR -- simple-example-default: Verify failed on instance <simple-example-default>.
E, [2019-08-05T17:55:44.166113 #11375] ERROR -- simple-example-default: ------Exception-------
E, [2019-08-05T17:55:44.166127 #11375] ERROR -- simple-example-default: Class: Kitchen::ActionFailed
E, [2019-08-05T17:55:44.166135 #11375] ERROR -- simple-example-default: Message: simple_example gcp: Can't find train plugin gcp. Please install it first.
E, [2019-08-05T17:55:44.166141 #11375] ERROR -- simple-example-default: ----------------------
E, [2019-08-05T17:55:44.166148 #11375] ERROR -- simple-example-default: ------Backtrace-------
E, [2019-08-05T17:55:44.166154 #11375] ERROR -- simple-example-default: /home/vagrant/.rbenv/versions/2.5.3/lib/ruby/gems/2.5.0/gems/kitchen-terraform-5.1.0/lib/kitchen/verifier/terraform.rb:109:in `rescue in call'
...
```

At this point I have no idea what I'm doing wrong/differently for train to not be able to find inspec-gcp...

**Comment by tdsacilowski** [comment]
Any just to be sure, I can confirm that this works with kitchen-terraform 4.8 (default for the https://github.com/terraform-google-modules/terraform-google-module-template project):

Gemfile:
```
ruby '2.5.3'

source 'https://rubygems.org/' do
  gem 'kitchen-terraform', '~> 4.8'
  # gem 'kitchen-terraform', '~> 5.1'
end
```

Verify output:
```
vagrant@ubuntu-xenial:~/terraform-google-test-module$ bundle exec kitchen verify
-----> Starting Kitchen (v1.25.0)
-----> Verifying <simple-example-gcp>...
$$$$$$ Running command `terraform workspace select kitchen-terraform-simple-example-gcp` in directory /home/vagrant/terraform-google-test-module/test/fixtures/simple_example
$$$$$$ Running command `terraform output -json` in directory /home/vagrant/terraform-google-test-module/test/fixtures/simple_example
simple_example local: Verifying
Skipping profile: 'inspec-gcp' on unsupported platform: 'ubuntu/16.04'.

Profile: simple_example
Version: (not specified)
Target:  local://

  ✔  gsutil: gsutil
     ✔  Command: `gsutil ls -p teddy-testing` exit_status should eq 0
     ✔  Command: `gsutil ls -p teddy-testing` stderr should eq ""
     ✔  Command: `gsutil ls -p teddy-testing` stdout should match "gs://simple-example-marmoset"
  ✔  gcloud: gcloud
     ✔  Command: `gcloud --project=teddy-testing services list --enabled` exit_status should eq 0
     ✔  Command: `gcloud --project=teddy-testing services list --enabled` stderr should eq ""
     ✔  Command: `gcloud --project=teddy-testing services list --enabled` stdout should match "storage-api.googleapis.com"


Profile Summary: 2 successful controls, 0 control failures, 0 controls skipped
Test Summary: 6 successful, 0 failures, 0 skipped
simple_example gcp: Verifying

Profile: simple_example
Version: (not specified)
Target:  gcp://teddy-terraform@teddy-testing.iam.gserviceaccount.com

  ✔  gcp: GCP Resources
     ✔  Bucket simple-example-marmoset should exist


Profile: Google Cloud Platform Resource Pack (inspec-gcp)
Version: 0.9.0
Target:  gcp://teddy-terraform@teddy-testing.iam.gserviceaccount.com

     No tests executed.

Profile Summary: 1 successful control, 0 control failures, 0 controls skipped
Test Summary: 1 successful, 0 failures, 0 skipped
       Finished verifying <simple-example-gcp> (0m5.32s).
-----> Kitchen is finished. (0m6.79s)
```

**Comment by tdsacilowski** [comment]
Ok, I think I have success here! Here was my logic:

1. First, as far as I can tell, inspec-gcp is not yet compatible with InSpec 4: see https://github.com/inspec/inspec-gcp/blob/6b2bfcd0847b69c5ddf6ac674dfab1adea95f0e7/Gemfile#L8 and https://github.com/inspec/inspec-gcp/issues/153

2. Which tells me that I need to pin to InSpec 3 when using Kitchen-Terraform v5.1: see https://github.com/newcontext-oss/kitchen-terraform/blob/master/CHANGELOG.md#changed. However, this wasn't working for me previously and I couldn't figure out why, that is...

3. Until I came across this InSpec issue: https://github.com/inspec/inspec/issues/3066. Adding `inspec_version: '~> 3.0'` to my `inspec.yml` file seemed to have done the trick of properly pinning to InSpec 3

4. I did, however, get the following error during `verify`, which I hadn't previously:
```
>>>>>> ------Exception-------
>>>>>> Class: Kitchen::ActionFailed
>>>>>> Message: 1 actions failed.
>>>>>>     Failed to complete #verify action: [Could not open library 'sodium': sodium: cannot open shared object file: No such file or directory.
Could not open library 'libsodium.so': libsodium.so: cannot open shared object file: No such file or directory] on root-example-terraform
>>>>>> ----------------------
>>>>>> Please see .kitchen/logs/kitchen.log for more details
>>>>>> Also try running `kitchen diagnose --all` for configuration
```

Running `sudo apt install -y libsodium-dev` resolved this for me and I now have inspec-gcp tests running:
```
vagrant@ubuntu-xenial:~/tfe-demo-modules/terraform-gcp$ bundle exec kitchen verify
-----> Starting Kitchen (v2.2.5)
-----> Verifying <root-example-terraform>...
$$$$$$ Running command `terraform workspace select kitchen-terraform-root-example-terraform` in directory /home/vagrant/tfe-demo-modules/terraform-gcp/examples/root-example
$$$$$$ Running command `terraform output -json` in directory /home/vagrant/tfe-demo-modules/terraform-gcp/examples/root-example
root-example local: Verifying
Skipping profile: 'root-example' on unsupported platform: 'ubuntu/16.04'.

Test Summary: 0 successful, 0 failures, 0 skipped
root-example gcp: Verifying

Profile: root-example
Version: (not specified)
Target:  gcp://teddy-terraform@teddy-testing.iam.gserviceaccount.com

  ✔  gcp_compute_instance_exists: Instance tfe
     ✔  Instance tfe status should eq "RUNNING"
     ✔  Instance tfe first_network_interface_nat_ip_exists should equal true
     ✔  Instance tfe first_network_interface_name should eq "external-nat"
     ✔  Instance tfe first_network_interface_type should eq "one_to_one_nat"


Profile: Google Cloud Platform Resource Pack (inspec-gcp)
Version: 0.11.0
Target:  gcp://teddy-terraform@teddy-testing.iam.gserviceaccount.com

     No tests executed.

Profile Summary: 1 successful control, 0 control failures, 0 controls skipped
Test Summary: 4 successful, 0 failures, 0 skipped
```

**Comment by bgeesaman** [comment]
I've found the same issue, and I tracked it down to the `rbnacl` gem that `kitchen-terraform` `5.+` pulls in that `kitchen-terraform 4.x` does not.  With `rbnacl` installed, I get the `train can't find plugin 'gcp'`.  Without it and with `kitchen-terraform 4.x` installed, `inspec-gcp` works just fine.

**Comment by aaron-lane** [comment]
@bgeesaman please post any information from your investigation to https://github.com/newcontext-oss/kitchen-terraform/issues/351.

**Pull request merged**